pkg/build/api/validation/validation.go

```diff
@@ -171,6 +178,10 @@ func validateSource(input *buildapi.BuildSource, isCustomStrategy, isDockerStrat
 		}
 	}
 
+	if input.Git == nil && isJenkinsPipelineStrategyFromRepo {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("git"), "", "must be set when using Jenkins Pipeline strategy with Jenkinsfile from a git repo"))
```

Comment:
There's a Jenkinsfile field on the JenkinsPipelineStrategy which would represent an in-line pipeline definition so you don't need a git repo.  Similar to the Dockerfile field.